### ISSUE TITLE
[Bug]: The quota has been exceeded

### ISSUE THREAD

**Issue by CoreyScogin** [remotely-save/remotely-save#469]
### What happened?

When attempting to sync a vault of 800 items I receive the error `error while generating_plan` then `The quota has been exceeded`.  Syncing a test vault with 200 items to this same device succeeds.

Syncing the same 800 item vault succeeds on Windows and macOS.

My test scenarios have included:
- S3
- Dropbox
- Encryption On
- Encryption Off

### What OS are you using?

iOS (iPhone, iPad)

### What remote cloud services are you using? (Please choose the specified one if it's in the list)

S3, Dropbox

### Version of the plugin

0.3.38

### Version of Obsidian

1.4.16

### Using password or not

- [ ] Yes.

### Ensure no sensitive information

- [X] I ensure that no sensitive information is submitted in the issue.

**Comment by fyears** [comment]
click the reset button in settings. maybe the cache database is too large

**Comment by CoreyScogin** [comment]
Thanks for the quick response!  
Clicking *Reset Local Internal Cache/Databases* does not resolve the issue. 
Clicking *Delete Sync Mappings History in DB* does resolve the issue but only for the next sync. Subsequent syncs consistently fail.

**Comment by fyears** [comment]
do you know how to open the dev console? maybe your obsidian's local storage is fullfilled by many caches (by other plugins)

**Comment by CoreyScogin** [comment]
There were a couple of other plugin cache databases in local storage. I reinstalled the app to get rid of those. The problem still remains. 
The only plugin I have added to Obsidian is this one.

Here's where Safari dev tools say the error is happening:
![image](https://github.com/remotely-save/remotely-save/assets/1843966/2502dc7f-04e1-4eb8-b00b-36cea89ed3fe)

**Comment by fyears** [comment]
what ios version are you using?

by default the plugin should use indexdb, not localstorage.

but clearly your error msg comes from localstorage: https://stackoverflow.com/questions/42467908/quotaexceedederror-dom-exception-22-the-quota-has-been-exceeded-on-safari-in

i have no idea yet but in next version i would try to explicitly set the storage into indexdb and avoid localstorage

**Comment by CoreyScogin** [comment]
It turns out Lockdown Mode is the culprit. I keep my device in Lockdown Mode. 

## Workaround 
Excluding Obsidian from Lockdown Mode in *Settings / Privacy & Security / Lockdown Mode / Configure Web Browsing* allows sync to complete.

On my device, the sync plan history is stored in Local Storage at `remotelysavedb/syncplanshistory/[id]`. When in Lockdown Mode, deleting this file allowed the sync to complete but only once. Subsequent syncs throw the quota error. 

You mentioned explicitly setting storage into indexdb. Do you think this would allow sync to work while in Lockdown Mode?
I know this is an edge case but if you're willing to try this I'd be happy to test it.

**Comment by fyears** [comment]
Thanks for your discovery!!! I use iPhone but never know Lockdown Mode before.

After a quick search, it seems that lockdown mode disable indexdb and localstorage. So right now it's clwar that the js library cannot detect indexdb and fallback to localstorage and still fail.

https://forum.obsidian.md/t/obsidian-sync-connecting-to-remove-vault-failing-on-ios-in-lockdown-mode/55546/19

https://github.com/WebKit/WebKit/pull/7506

I think the only workaround is using files, instead of indexdb or localstorage, to bypass the limitation. Which is technically possible but admittedly I would keep it at low priority.

**Comment by CoreyScogin** [comment]
Thanks! I understand. I'll mark this closed/not resolved.

**Comment by pengyuwei** [comment]
When using webdav for file synchronization, the Lockdown Mode of the operating system is turned off. During synchronization, an error message is received stating 'the quota has been exceeded', indicating that the file has not been synchronized.
